design/app-to-app.md

```diff
@@ -0,0 +1,87 @@
+App to App Interactions
+===========
+
+This document describes an extension to the SDK to allow MRE apps to receive content (e.g., photos) from the MRE clients. Content may be generated by the client itself, or, in future iterations not yet described here, may arrive from other apps with coordination from a particular client.
+
+## API Design
+This design introduces the concept of a `SharingMessage`, a base class of content-specific messages, such as `PhotoSharingMessage`. We provide an outline for the implementation of `PhotoSharingMessage` only, but expect to add more sharing capabilities over time.
+
+`SharingMessage` is itself relatively straightforward, but contains enough information to allow the recipient (callback listener) to cast it to the appropriate type (e.g., `PhotoSharingMessage`). We envision multiple message types and don't want to add a registration api for each.
+
+```ts
+class SharingMessage {
+    messageType: Readonly<string>;
+    sessionId: Readonly<string>;
```

Comment:
Session ID is redundant here, because it is already associated with the websocket that the message arrives on.

Reply:
The reason for including it here is that a callback instance might be registered across sessions.